### ISSUE TITLE
Change L007 to specify end of line as well as beginning of line consistency

### ISSUE THREAD

**Issue by tunetheweb** [sqlfluff/sqlfluff#1261]
L007 enforces "Operators near newlines should be after, not before the newline."

That is this:

```sql
SELECT
    a
    + b
FROM foo
```

is preferred over this:

```sql
SELECT
    a +
    b
FROM foo
```

I'd like to suggest an `operator-before-end-of-line` config param (defaulted to `False` for backwards compatibility) which can be used the consistently enforce the latter behaviour for teams that prefer that.

If accepted, we should reword this rule to "Operators near newlines should be consistent in regards to placement before or after the newline (default is to prefer before)."

Happy to look at this one myself.

**Comment by barrywhart** [comment]
This sounds good to me -- I could see a preference for either style. 👍🏽
